src/analysis/processing/qgsalgorithmlayoutatlastopdf.cpp

```diff
@@ -248,5 +256,182 @@ QVariantMap QgsLayoutAtlasToPdfAlgorithm::processAlgorithm( const QVariantMap &p
   return outputs;
 }
 
+// QgsLayoutAtlasToMultiplePdfAlgorithm
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::name() const
+{
+  return QStringLiteral( "atlaslayouttomultiplepdf" );
+}
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::displayName() const
+{
+  return QObject::tr( "Export atlas layout as PDF (multiple files)" );
+}
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Exports an atlas layout to multiple PDF files." );
+}
+
+QString QgsLayoutAtlasToMultiplePdfAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm outputs an atlas layout to multiple PDF files.\n\n"
+                      "If a coverage layer is set, the selected layout's atlas settings exposed in this algorithm "
+                      "will be overwritten. In this case, an empty filter or sort by expression will turn those "
+                      "settings off.\n"
+                    );
+}
+
+void QgsLayoutAtlasToMultiplePdfAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  QgsLayoutAtlasToPdfAlgorithmBase::initAlgorithm();
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "OUTPUT_FILENAME" ), QObject::tr( "Output filename" ), QString( "'output_'||@atlas_featurenumber" ), QString(), false ) );
```

Comment:
This should be optional and default to blank in order to allow users to use the expression stored within the layout.
```suggestion
  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "OUTPUT_FILENAME" ), QObject::tr( "Output filename" ), QString(), QString(), true ) );
```